cmd/mapper/mapper.go

```diff
@@ -137,3 +222,35 @@ func main() {
 
 	<-forever
 }
+
+// schemaFromDatasetOperation returns the operation done with dataset
+// supplied in body of the message
+func schemaFromDatasetOperation(body []byte) (string, error) {
+	message := make(map[string]interface{})
+	err := json.Unmarshal(body, &message)
+	if err != nil {
+		return "", err
+	}
+
+	datasetMessageType, ok := message["type"]
+	if !ok {
+		return "", errors.New("Malformed message, dataset message type is missing")
+	}
+
+	datasetOpsType, ok := datasetMessageType.(string)
+	if !ok {
+		return "", errors.New("Could not cast operation attribute to string")
+	}
+
+	switch datasetOpsType {
+	case "mapping":
```

Comment:
`mapping` or `mappings`

Reply:
`mapping` as it is not queue name `mappings`

Reply:
this gets the json schema corresponding with the type found in the message